http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPResponseListener.java

```diff
@@ -0,0 +1,144 @@
+//                            future.addListener(ChannelFutureListener.CLOSE);
+//                        }
+//                        if (HTTPTransportContextHolder.getInstance().getHandlerExecutor() != null) {
+//                            HTTPTransportContextHolder.getInstance().getHandlerExecutor().
+//                                    executeAtSourceResponseSending(httpMessage);
+//                        }
+//                        break;
+//                    }
+//                    ctx.write(httpContent);
+//                }
+//            }
+//            else if (httpMessage instanceof DefaultCarbonMessage) {
+//                DefaultCarbonMessage defaultCMsg = (DefaultCarbonMessage) httpMessage;
+//                if (defaultCMsg.isEndOfMsgAdded() && defaultCMsg.isEmpty()) {
+//                    ChannelFuture future = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+//                    if (connectionCloseAfterResponse) {
+//                        future.addListener(ChannelFutureListener.CLOSE);
+//                    }
+//                    return;
+//                }
+//                while (true) {
+//                    ByteBuffer byteBuffer = defaultCMsg.getMessageBody();
+//                    ByteBuf bbuf = Unpooled.wrappedBuffer(byteBuffer);
+//                    DefaultHttpContent httpContent = new DefaultHttpContent(bbuf);
+//                    ctx.write(httpContent);
+//                    if (defaultCMsg.isEndOfMsgAdded() && defaultCMsg.isEmpty()) {
+//                        ChannelFuture future = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+//                        if (HTTPTransportContextHolder.getInstance().getHandlerExecutor() != null) {
+//                            HTTPTransportContextHolder.getInstance().getHandlerExecutor().
+//                                    executeAtSourceResponseSending(httpMessage);
+//                        }
+//                        if (connectionCloseAfterResponse) {
+//                            future.addListener(ChannelFutureListener.CLOSE);
+//                        }
+//                        break;
+//                    }
+//                }
+//            }
+        }
+    }
+
+    // Decides whether to close the connection after sending the response
+    private boolean shouldConnectionClose(CarbonMessage responseMsg) {
+        String responseConnectionHeader = responseMsg.getHeader(Constants.HTTP_CONNECTION);
+        String requestConnectionHeader = requestDataHolder.getConnectionHeader();
+        if ((responseConnectionHeader != null &&
+                Constants.CONNECTION_CLOSE.equalsIgnoreCase(responseConnectionHeader))
+                || (requestConnectionHeader != null &&
+                Constants.CONNECTION_CLOSE.equalsIgnoreCase(requestConnectionHeader))) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+
```

Comment:
Do we need to handle any logic here for error scenario on registered listener? Was it intentional to left this as empty?